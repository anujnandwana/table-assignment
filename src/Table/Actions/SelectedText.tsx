import React from "react";

interface SelectedTextProps {
    selectedRowIds: string[]
}

const SelectedText: React.FC<SelectedTextProps> = ({ selectedRowIds }) => {
    return (
        <span className="selected-text" role="selected rows">
            {!selectedRowIds.length
                ? "None Selected"
                : `Selected ${selectedRowIds.length}`}
        </span>
    );
};

export default SelectedText;
