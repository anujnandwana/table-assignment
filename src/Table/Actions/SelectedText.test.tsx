import SelectedText from './SelectedText';
import { mount } from 'enzyme';

describe('SelectedText component', () => {
  const selectedRowIds = ["mockedId"];

  test('SelectedText component should match snapshot', () => {
    const component = mount(<SelectedText selectedRowIds={selectedRowIds} />);
    expect(component).toMatchSnapshot();
  });

  test('select count should show Selected 1 when 1 row is selected ', () => {
    const component = mount(<SelectedText selectedRowIds={selectedRowIds} />);
    const selectedText = component.find("span.selected-text");

    expect(selectedText.at(0).text()).toBe("Selected 1");
  });

  test('select count should show None Selected when 0 row is selected ', () => {
    const component = mount(<SelectedText selectedRowIds={[]} />);
    const selectedText = component.find("span.selected-text");

    expect(selectedText.at(0).text()).toBe("None Selected");
  });

});
