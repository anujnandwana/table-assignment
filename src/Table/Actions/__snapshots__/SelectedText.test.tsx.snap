// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedText component SelectedText component should match snapshot 1`] = `ReactWrapper {}`;
